[raz] Add JWT support for Hue-Raz Integration

Harshg999 opened the pull request:
## What changes were proposed in this pull request?

- Instead of directly asking RAZ for delegation token via Kerberos credentials, we are now fetching JWT from Knox host (this setup can be in HA mode) and then serve that JWT to RAZ to get back the delegation token.
- The later flow using the delegation token remains the same.

## How was this patch tested?

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

Harshg999 commented:
PR is ready for review.

wing2fly commented on desktop/core/src/desktop/conf.py:
```diff
@@ -2174,6 +2174,28 @@ def has_raz_url():
+
+def is_sdxaas_jwt_enabled():
+  """Check if SDXaaS token url is configured"""
+  return SDXAAS.TOKEN_URL.get() != ''
```
Nit: Take a look this one: https://stackoverflow.com/questions/9573244/how-to-check-if-the-string-is-empty

Harshg999 replied:
Changed it and using `bool()` now.

ranade1 reviewed (approved):
Looks good to me. Have you tested it end-to-end?

ranade1 reviewed (approved):
Looks good to me. Have you tested it end-to-end?

Harshg999 commented:
> * Updated existing UTs and added new UTs.
> * Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
> * However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

Harshg999 commented:
> Looks good to me. Have you tested it end-to-end?

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.
